src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -336,69 +334,71 @@ def _check_remote_source_code(source_location):
             else:
                 raise CLIError("'{}' doesn't exist.".format(source_location))
 
-    raise CLIError(
-        "'{}' is not a valid remote url for git or tarball.".format(source_location))
+    raise CLIError("'{}' is not a valid remote url for git or tarball.".format(source_location))
 
 
 def _upload_source_code(client, registry_name, resource_group_name, source_location, tar_file_path, docker_file_path):
+    logger.debug("Starting to acquire the access token to upload the source code.")
+    ignore_list = _load_dockerignore_file(source_location)
+    common_vcs_ignore_list = {'.git', '.gitignore', '.bzr', 'bzrignore', '.hg', '.hgignore', '.svn'}
 
-    try:
-        logger.debug(
-            "Starting to acquire the access token to upload the source code.")
-
-        source_upload_location = client.get_build_source_upload_url(
-            resource_group_name=resource_group_name, registry_name=registry_name)
-
-        ignore_list = _load_dockerignore_file(source_location)
+    def _filter_file(tarinfo):
```

Comment:
I would prefer moving this out to make it more readable and reusable (but kept as is for now), unless there are other concerns to inline this method. The performance difference is minimal. /cc @ehotinger @northtyphoon

Reply:
I said the same thing before, @northtyphoon preferred it kept inline. Doesn't matter to me, it's isolated to this function anyways right now so there's no real benefit to breaking it out.